### ISSUE TITLE
Source maps show wrong lines

### ISSUE THREAD

**Issue by romeovs** [hapijs/lab#338]
I'm using `lab` for a small project and it's working very good.

I'm using running the test coverage with a custom `babel` transform (taken from the example
of the README):

```js
'use strict';
var Babel = require('babel');

module.exports = [
  {
    ext: '.js'
  , transform: function (content, filename) {
      // Make sure to only transform your code or the dependencies you want
      if (filename.indexOf('node_modules') === -1) {

        var opts  = {
          sourceMap: 'inline'
        , filename: filename
        , sourceFileName: filename
        , stage: 0
        , optional: [
            'runtime'
          ]
        };

        var result = Babel.transform(content, opts);
        return result.code;
      }

      return content;
    }
  }
];
```

The displaying of the uncovered lines seems off though, an seems to be shifted by 6 lines (as can be seen [here](https://coveralls.io/builds/2291010/source?filename=src%2Fimmutable.js)).

What can I do to fix this?

**Comment by ldesplat** [comment]
I does not look like the Lab's lcov reporter takes the sourcemaps output into consideration like the console or html reporter.

It should be looking at line.originalLine amongst other things.

**Comment by Marsup** [comment]
The files are also wrong in the console reporter since a few releases.

**Comment by romeovs** [comment]
@ldesplat I also have the same problem with the `html` reporter.

**Comment by ldesplat** [comment]
@romeovs Drag your browser page from the html report to the right. There are two extra columns that show the file it comes from and the real line number. I agree the html report is not optimized for the transform use case where there is a 1 to 1 mapping from source to transformed.

**Comment by romeovs** [comment]
@ldesplat I do not see what you mean. This is what I see:
![screen shot 2015-04-09 at 01 04 27](https://cloud.githubusercontent.com/assets/1250185/7057279/6a79a01c-de54-11e4-8aca-25ab89b3d3a7.png)

Do you mean the rightmost column?

I'm talking about the fact that line 41 is marked for example, while there is clearly not much to test there.

**Comment by ldesplat** [comment]
@romeovs Yes, that's what I meant. Indeed it highlights line 41 for example, but then it says it's actually line 36 on the right. So it is reject(err); that is not covered.

That's what I meant by the html reporter not being very good for this use case.

**Comment by romeovs** [comment]
Ok maybe it would be nice to have an option to make the red lines over the original code? This is much more intuitive to me since to code shown is the original code, not the compiled code.

Also how should I interpret the following:
![screen shot 2015-04-09 at 01 22 47](https://cloud.githubusercontent.com/assets/1250185/7057558/2df52316-de57-11e4-8f6a-9a5184b6ffc3.png)

Here you see line 10 before line 1 and then a lot of 1's and 2's repeated. I don't get this since the code is copied verbatim from the original, so it doesn't seem correct.

**Comment by ldesplat** [comment]
Yeah, I agree the html report needs some modifications to make it support that use case.

I have no idea how to help you with this last screenshot. I have seen similar behavior but it was not as glaring as what you have there. It's obviously looking like a bug but I don't know where it comes from.

**Comment by geek** [comment]
@romeovs the fix for the line numbers is now published in lab 5.6.0.  What is still outstanding with this issue that we need to fix?

**Comment by romeovs** [comment]
On `5.7.0` it now just shows the transformed code in the html reporter, which is also not what we want I think.

However, I do not see directly how it can be done differently.

**Comment by ldesplat** [comment]
@romeovs Yes, it was the easiest way of making it behave at least a bit better.

The way the live transforms are done right now, is there is 1 source file for 1 output file. But a lot of tools, may actually move code around or put file together. It makes it a lot harder to show just the original code in the reporters. So, if we add specific logic to the transform path in lab in the reporters then we could show the original code and also do lcov sourcemap reporting. But it seems to add a lot of special logic which could be brittle.

Because of this and other reasons, I am now using a build step to transpile all my js code instead of doing it auto-magically everywhere.

**Comment by garthk** [comment]
TL;DR:

* If you use `--transform` and `--reporter html`, you'll see transformed code
* `--sourcemaps` doesn't help

Might be worth documenting that, given no progress in nine months.

**Comment by Marsup** [comment]
Bugs are already documented in the form of labeled opened issues, if you have time to spend, better spend it fixing the bug.

**Comment by geek** [comment]
I was not able to reproduce with the latest today.
